### DISCUSSION TITLE
how to generate file dumpwave by cocotb

### DISCUSSION THREAD

**Discussion by AnhHonag** [cocotb/cocotb#4134] [Q&A] [answered]
in systemverilog to dump wave file use bellow:
    initial begin
//      $display("input from test a=%b,b=%b,cin=%b",pif.a,pif.b,pif.cin);
        $dumpfile("dump.vcd");
        $dumpvars(1);
    end

in cocotb, how will it be written?

**Comment by ktbarrett** [comment, marked as answer]
If you are using the Makefiles, set the variable `WAVES=1`. If you are using the Python runner, use the `waves=True` argument to both the `build()` and `test()` method calls.

**Reply by Jax438** [reply]
And how we should do it if integrate cocotb into the classical simulation flow as mentioned in https://github.com/cocotb/cocotb/discussions/3839#discussioncomment-9084144 ?

**Reply by ktbarrett** [reply]
@Jax438 It depends on your simulator. Look through the Makefiles to see what command line options `WAVES` is setting. [For example](https://github.com/cocotb/cocotb/blob/4d66e32e0ef035ebccb285c5a1ead11805c6c060/src/cocotb_tools/makefiles/simulators/Makefile.questa#L118-L120).

**Reply by Jax438** [reply]
Thanks for the reference. At the end, in my case, I have added the following in my design compilation .do file:
```
vcd file waveforms.vcd
vcd add -r /*
```
Full details here: https://github.com/cocotb/cocotb/discussions/4692#discussioncomment-13224979
